lib/provider.py

```diff
@@ -107,26 +121,41 @@ def perform_search(search_type, data):
         logging.warning("No scrapers configured/enabled")
         return None
 
+    if not isinstance(search_types, list):
+        search_types = [search_types]
+
     runner_class = ProgressScraperRunner if get_boolean_setting("enable_bg_dialog") else ScraperRunner
     with runner_class(scrapers, num_threads=get_int_setting("thread_number")) as runner:
-        runner_data = runner.parse_query(data) if search_type == "query" else runner.parse(search_type, data)
-
-        for scraper, scraper_results in runner_data:
-            logging.debug("Processing %s scraper results", scraper.name)
-            for scraper_result in scraper_results:
-                magnet = Magnet(scraper_result["magnet"])
-                try:
-                    info_hash = magnet.parse_info_hash()
-                except InvalidMagnet:
-                    continue
-                if info_hash == "0" * 40:
-                    continue
-
-                magnet_result = results.get(info_hash)  # type: Result
-                if magnet_result is None:
-                    results[info_hash] = Result(scraper, scraper_result)
-                else:
-                    magnet_result.add_result(scraper, scraper_result)
+        for search_type in search_types:
```

Comment:
This can't be done this way, as it breaks the progress dialog (you are, when including seasons, performing the search twice, so the progress will go up to 200%

Have a look at:
https://github.com/i96751414/script.flix.magneto/blob/820e30f25773d2b82900b05d293268ce07a27125/lib/provider.py#L144-L146

and:
https://github.com/i96751414/script.flix.magneto/blob/820e30f25773d2b82900b05d293268ce07a27125/lib/scraper.py#L201-L212